### DISCUSSION TITLE
Struct/list unpacking and `*COLUMNS`

### DISCUSSION THREAD

**Discussion by soerenwolfers** [duckdb/duckdb#14224] [General]
Motivated by https://github.com/duckdb/duckdb/discussions/14188:
 
Is there a way to unpack structs and lists/arrays similar to how `*COLUMNS` unpacks a `COLUMNS` expression?

I don't think `foo(*regular_list_or_struct_column)` can work unfortunately because then `foo(*COLUMNS(...))` would be ambiguous. 

Maybe the ship has sailed on this, but wouldn't it be better to be have the ability to get `struct_pack(*COLUMNS(...))` differently (e.g. via `table_alias.select_regexp('regexp')`, see #13840) and then have the separate ability to unpack lists and structs via `*` and/or `**`?

**Comment by Tishj** [comment]
That would be awesome, but I don't think that's possible.

`*COLUMNS` works because it is resolved at bind time
we have an expression, that expression has a `*COLUMNS` expression as child (or one of its children)

While binding, we can resolve the `*COLUMNS` and replace it with the expression(s) it produces.
To be able to have any arbitrary list expand, it would have to be resolvable at bind time (which is only possible for constants, perhaps a prepared parameter also works) which is very limited and would only be confusing and cause people to ask us to lift this unliftable restriction (for evidence of this, think of the table-function-only-allowing-constant-parameters problem)

**Comment by frankjar** [comment]
I don't know if it can be useful for your case, but i have found a solution to solve my problem.

I have struct fields and i would like to extract specific entry for all the columns into a list.
First step : use *columns() like this `select [*columns(*)] from my_table;`
Next step: use list_transform() like this `select list_transform( [*columns(*)], s->s.entry) from my_table`

May be help
